File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans:List[List[int]] = []
        ans.append([1])

        for i in range(0,numRows-1):
            prev:int  = 0
            arr:List[int] = []

            n:int = len(ans[i])

            for j in range(0,n):
                arr.append(prev + ans[i][j])
                prev = ans[i][j]

            next = ans[i][n-1]
            arr.append(next)
            ans.append(arr)
        
        return ans
        

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {

        vector<vector<int>> ans;
        ans.push_back({1});

        for(int i=0;i<numRows-1;i++)
        {
            int prev = 0;
            vector<int> arr;

            for(int j=0;j<ans[i].size();j++)
            {
                arr.push_back(ans[i][j] + prev);
                prev = ans[i][j];
            }
            int next = ans[i][ans[i].size() - 1];
            arr.push_back(next);
            ans.push_back(arr);
        }

        return ans;
        
    }
};